lib/Runtime/PlatformAgnostic/Platform/Windows/UnicodeText.cpp

```diff
@@ -399,18 +399,29 @@ namespace PlatformAgnostic
                 Js::WindowsGlobalizationAdapter* globalizationAdapter = threadContext->GetWindowsGlobalizationAdapter();
                 Js::DelayLoadWindowsGlobalization* globLibrary = threadContext->GetWindowsGlobalizationLibrary();
                 HRESULT hr = globalizationAdapter->EnsureDataTextObjectsInitialized(globLibrary);
-                // Failed to load windows.globalization.dll or jsintl.dll. No unicodeStatics support
-                // in that case.
+                // Failed to load windows.globalization.dll or jsintl.dll. No unicodeStatics support in that case.
                 if (SUCCEEDED(hr))
                 {
                     auto winGlobCharApi = globalizationAdapter->GetUnicodeStatics();
                     if (winGlobCharApi != nullptr)
                     {
                         return true;
                     }
+                    else
+                    {
+                        // did not find winGlobCharApi
+                        Js::Throw::FatalInternalGlobalizationError();
+                    }
+                }
+                else
+                {
+                    // failed to initialize Windows Globalization
+                    Js::Throw::FatalInternalGlobalizationError();
                 }
 
-                return false;
+#ifndef DBG
+                return false; // in debug builds, this is unreachable code
```

Comment:
This is also unreachable in release, no? Also, you could take away both elses and just have a single throw at the end here, before returning the unreachable value.